enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,317 @@
+                  azure:
+                    properties:
+                      resourceTags:
+                        description: resourceTags is a list of additional tags to apply to Azure
+                        resources created for the cluster. See 
+                        https://docs.microsoft.com/en-us/rest/api/resources/tags for information 
+                        on tagging Azure resources. Azure supports a maximum of 50 tags per 
+                        resource except for few, which have limitation of 15 tags. OpenShift 
+                        reserves 5 tags for its internal use, and allows 10 tags 
+                        for user configuration.
+                        type: array
+                        maxItems: 10
+                        items:
+                          description: AzureResourceTag is a tag to apply to Azure resources created for the cluster.
+                          type: object
+                          required:
+                            - key
+                            - value
+                          properties:
+                            key:
+                              description: key is the key of the tag
+                              type: string
+                              maxLength: 128
+                              minLength: 1
+                              pattern: ^[a-zA-Z][0-9A-Za-z_.=+-@]+$
+                            value:
+                              description: value is the value of the tag.
+                              type: string
+                              maxLength: 256
+                              minLength: 1
+                              pattern: ^[0-9A-Za-z_.=+-@]+$
+```
+
+### Implementation Details/Notes/Constraints [optional]
+Add a new field `resourceTags` to `.status.platformStatus.azure` of the 
+`infrastructure.config.openshift.io` type. Tags included in the `resourceTags` field 
+will be applied to new resources created for the cluster by the in-cluster operators.
+
+The `resourceTags` field in `status.platformStatus.azure` of `infrastructure.config.openshift.io`
+will be populated by the installer using the entries from `platform.azure.userTags` field of `install-config`.
+
+`status.platformStatus.azure` of `infrastructure.config.openshift.io` is immutable.
+
+All operators that create Azure resources will apply these tags to all Azure 
+resources they create. List of in-cluster operators managing cloud resources
+could vary across platform types, example for AWS there are additional operators
+like aws-ebs-csi-driver-operator, aws-efs-csi-driver-operator to manage specific
+resources.
+
+| Operator | Resources created by the operator |
+| -------- | ----------------------------- |
+| cloud-network-config-controller | Private IP address |
+| cluster-image-registry-operator | Storage Account |
+| cluster-ingress-operator | Load Balancer, DNS records |
+| cloud-credential-operator | IAM roles and policies |
```

Comment:
Enhancement looks good wrt CCO which does not presently have Mint mode for Azure due to the deprecation of the graph API (see https://github.com/openshift/cloud-credential-operator/pull/433) and would not create resources. However if CCO were to create resources, CCO could check the infrastructure object to get the list of tags.

There is an upcoming request for short-lived credential support on Azure and the IAM resources would be created by CCO's command-line utility (ccoctl) ahead of installation (see https://github.com/openshift/enhancements/pull/1301). Is there a need to provide a way to tag these resources? CCO would not have access to the install-config when creating this infrastructure.

Reply:
My bad, I missed this while doing code walk through for the enhancement. CCO reconciles just the credentials secret but does not create or update any credentials.

/cc @TrilokGeer
Customer feedback was to have all possible resources, created for OCP, to be tagged. But since resources are created before cluster creation, should ccoctl provide options for tags with azure identity resource creation related sub-commands?

Reply:
@abutcher @bharath-b-rh , looking at https://github.com/openshift/enhancements/pull/1301, there'd be a need to have capabilities to tag the managed identities created by command-line utility (ccoctl). Assuming the changes in ccoctl are specific to OpenShift, it is suggested to have default tag ("kubernetes.io/cluster/${var.cluster_id}" = "owned") added by ccoctl. Though there are some scenarios which have limitations, it is detailed at https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-support#microsoftmanagedidentity.